tasks/service/async/src/test/java/org/projectnessie/nessie/tasks/async/BaseTasksAsync.java

```diff
@@ -0,0 +1,258 @@
+        async.call(
+            () -> {
+              throw new RuntimeException("hello");
+            });
+
+    Throwable mappedFailure =
+        stage.handle((result, failure) -> failure).toCompletableFuture().get();
+
+    soft.assertThat(mappedFailure)
+        .isInstanceOf(CompletionException.class)
+        .extracting(Throwable::getCause)
+        .isInstanceOf(RuntimeException.class)
+        .extracting(Throwable::getMessage)
+        .isEqualTo("hello");
+  }
+
+  @Test
+  public void scheduledExceptionallyWorks() throws Exception {
+    TasksAsync async = tasksAsync();
+
+    CompletionStage<Void> stage =
+        async.schedule(
+            () -> {
+              throw new RuntimeException("hello");
+            },
+            Instant.now());
+
+    Throwable mappedFailure =
+        stage.handle((result, failure) -> failure).toCompletableFuture().get();
+
+    soft.assertThat(mappedFailure)
+        .isInstanceOf(CompletionException.class)
+        .extracting(Throwable::getCause)
+        .isInstanceOf(RuntimeException.class)
+        .extracting(Throwable::getMessage)
+        .isEqualTo("hello");
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {1, 10, 1000})
+  public void scheduleWorks(int num) throws InterruptedException {
+    TasksAsync async = tasksAsync();
+
+    Semaphore sem = new Semaphore(0);
+
+    for (int i = 0; i < num; i++) {
+      async.schedule(sem::release, async.clock().instant());
+    }
+
+    soft.assertThat(sem.tryAcquire(num, 10, TimeUnit.SECONDS)).isTrue();
+  }
+
+  @Test
+  public void cancelDoesNotError() {
+    // See below
+    TasksAsync async = tasksAsync();
+    AtomicBoolean mark = new AtomicBoolean();
+    CompletionStage<Void> handle =
+        async.schedule(() -> mark.set(true), async.clock().instant().plus(1500, ChronoUnit.MILLIS));
+    handle.toCompletableFuture().cancel(true);
+    soft.assertThat(mark).isFalse();
+  }
+
+  @Test
+  public void cancelReallyWorks() throws InterruptedException {
+    TasksAsync async = tasksAsync();
+
+    CountDownLatch started = new CountDownLatch(1);
+    CountDownLatch cancelled = new CountDownLatch(1);
+
+    CompletionStage<Void> handle =
+        async.schedule(
+            () -> {
+              try {
+                started.countDown();
+                new Semaphore(0).acquire();
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+            },
+            async.clock().instant().plus(60, ChronoUnit.SECONDS));
+    soft.assertThat(handle).isNotNull();
+
+    AtomicReference<Object> result = new AtomicReference<>();
+    AtomicReference<Throwable> error = new AtomicReference<>();
+    handle.whenComplete(
+        (r, t) -> {
+          cancelled.countDown();
+          result.set(r);
+          error.set(t);
+        });
+
+    handle.toCompletableFuture().cancel(true);
+    soft.assertThat(cancelled.await(10, TimeUnit.SECONDS)).isTrue();
```

Comment:
Hmm this is only testing that `CompletionStage.whenComplete` works as designed. It's not testing that the runnable submitted to `async.schedule` was unscheduled / cancelled / interrupted. I think that, short of better options for Vertx, we could delete this test altogether and keep the Java-specific one in its own test class. Wdyt?